metadataserver/server.go

```diff
@@ -220,7 +233,13 @@ func (ms *MetadataServer) localIPV4(w http.ResponseWriter, r *http.Request) {
 
 func (ms *MetadataServer) publicIPV4(w http.ResponseWriter, r *http.Request) {
 	metrics.PublishIncrementCounter("handler.publicIPV4.count")
-	if _, err := fmt.Fprint(w, ms.ipv4Address); err != nil {
+	if ms.publicIpv4Address == nil {
```

Comment:
I wonder if this will break anyone.

Reply:
This endpoint gets called a *lot* by eureka-java-client, but not much otherwise - I'm going to check with them before rolling this out.